src/integrationTest/java/com/mongodb/spark/sql/connector/read/AbstractMongoStreamTest.java

```diff
@@ -598,7 +721,7 @@ private final void testStreamingQuery(
       final MongoConfig mongoConfig,
       final Consumer<MongoConfig> setup,
       final Consumer<MongoConfig>... consumers) {
-    testStreamingQuery(writeFormat, mongoConfig, DEFAULT_SCHEMA, null, setup, consumers);
+    testStreamingQuery(writeFormat, mongoConfig, DEFAULT_SCHEMA, null, null, setup, consumers);
```

Comment:
The changes below are test infrastructure related eg adding stream listener support and / or simplify the testing.